Packs/CybelAngel/Integrations/CybelAngel/CybelAngel.py

```diff
@@ -0,0 +1,689 @@
+
+    # get the service API url
+
+   # Get Cybelangel credentials
+    client_id = demisto.params().get('client_id')
+    client_secret = demisto.params().get('client_secret')
+    tenant_id = demisto.params().get('tenant_id')
+
+    # Get last run
+    last_run = demisto.getLastRun().get('start_time')
+
+    # Manage first fetch from client
+    first_fetch_interval = int(demisto.params().get('first_fetch'))
+    first_fetch = False if "first_pull" in demisto.getIntegrationContext().keys() else True
+
+    # Get token and token time if it exists
+    auth_token = demisto.getIntegrationContext().get('token')
+    expiry = demisto.getIntegrationContext().get('expiry')
+
+    # demisto.info(f'Token information ***** Time: {expiry} **** Token: {auth_token}')
+
+    demisto.debug('Command being called is {}'.format(demisto.command()))
+    try:
+
+        args = demisto.args()
+        # Create client connection to CybelAngel API
+        client = Client(
+            client_id=client_id,
+            client_secret=client_secret,
+            auth_token=auth_token,
+            token_time=expiry
+        )
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client)
+            return_results(result)
+
+            # return_results(result)
+
+        if demisto.command() == "fetch-incidents":
+            fetch_incidents(client, first_fetch=first_fetch, last_run=last_run, first_fetch_interval=first_fetch_interval)
+        elif demisto.command() == 'cybelangel-get-report-by-id':
+            return_results(get_report_by_id_command(client, args))
+        elif demisto.command() == 'cybelangel-get-report-attachment':
+            return_results(get_report_attachment_command(client, args))
+        elif demisto.command() == 'cybelangel-remediate':
+            return_results(remediate_command(client, args))
+        elif demisto.command() == 'cybelangel-get-comments':
+            return_results(get_comments_command(client, args))
+        elif demisto.command() == 'cybelangel-post-comment':
+            return_results(post_comment_command(client, tenant_id, args))
+        elif demisto.command() == 'cybelangel-update-status':
+            return_results(update_status_command(client, args))
+        elif demisto.command() == 'cybelangel-get-report-pdf':
+            return_results(get_report_pdf_command(client, args))
+        # TODO: REMOVE the following dummy command case
+        # elif demisto.command() == 'CybelAngel-get-report-by-id':
+        #     return_results(get_report_by_id_command(client,report_id=demisto.args()))
+
+        if demisto.command() == "test_command":
+            test_command(client)
+
+    # Log exceptions and return errors
+    except Exception as e:
+        return_error('Failed to execute {} command.\nError:\n{}'.format(demisto.command(), str(e)))
```

Comment:
```suggestion
        return_error(f'Failed to execute {demisto.command()} command.\nError:\n{str(e)}')

```

Reply:
The f-string provides better performance and readability, and it is the modern and most efficient way to work with strings.